### PR TITLE
(PE-8560) Adds more unit and integration tests.

### PR DESCRIPTION
**Pull request by kbrezina**
New unit and integration tests were added.

### CODE REVIEW

File: test/clj_puppetdb/core_test.clj [hunk 1]
```diff
@@ -26,4 +28,64 @@
...
+(deftest lazy-query-test
+  (testing "Should automatically fetch three pages by two items"
+    (let [data [["A" "B"] ["C" "D"] ["E" "F"]]]
+      (with-redefs [clj-puppetdb.http/GET (fn[_ _ {:keys [offset]}]
+                                            (let [idx (/ offset 2)]
+                                              (when (< idx 3) (nth data idx))))]
+        (let [seq (lazy-query nil nil {:limit 2 :order-by nil})]
+          (is (= (first seq) "A"))
+          (is (= (nth seq 4) "E"))
+          (is (= seq (flatten data)))))))
```
Comment (mruzicka) [inline test/clj_puppetdb/core_test.clj]: The first two assertions must be satisfied if the third one is, i.e. the first two are superfluous.

File: test/clj_puppetdb/core_test.clj [hunk 2]
```diff
@@ -26,4 +28,64 @@
...
+  (testing "Should automatically fetch two pages by tree items"
+    (let [data [["A" "B" "C"] [ "D" "E" "F"]]]
+      (with-redefs [clj-puppetdb.http/GET (fn[_ _ {:keys [offset]}]
+                                            (let [idx (/ offset 3)]
+                                              (when (< idx 2) (nth data idx))))]
+        (let [seq (lazy-query nil nil {:limit 3 :order-by nil})]
+          (is (= (first seq) "A"))
+          (is (= (nth seq 4) "E"))
+          (is (= seq (flatten data))))))))
```
Comment (mruzicka) [inline test/clj_puppetdb/core_test.clj]: Ditto as above: The first two assertions are superfluous.

**Comment by dmcpl** [comment]
I have a couple of small additions to make after my code has been submitted for the new functionality but this looks good to me!

**Comment by mruzicka** [comment]
LGTM

**Pull request merged**